### PR TITLE
Fix lint related to assigned but unused variables

### PR DESCRIPTION
**Pull request by bz2**
Resolves F841.

Also improves several tests to assert on relevant behaviours.

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -316,6 +316,7 @@ def test_invalid_names_raise(self):
 
     def test_empty_labels_list(self):
         h = Histogram('h', 'help', [], registry=self.registry)
+        h.observe(0)
```
Comment (brian-brazil) [inline tests/test_core.py]: This changes the test

Comment (bz2) [reply tests/test_core.py]: It does, but the test still fails if the change from commit e5e73e52 is reversed. It's really testing the constructor not recorded values, which is covered by `TestHistogram.test_histogram`.

Is there a better way of asserting "this isn't a `_LabelWrapper` that needs `labels()` called before using? I don't really want to use `isinstance()`.

Comment (brian-brazil) [reply tests/test_core.py]: You could not assign to a variable

Comment (bz2) [reply tests/test_core.py]: Okay, I wanted a better assertion but if you're fine with just checking the registry.

**Pull request merged**